### ISSUE TITLE
[lambda] Lambda Layers in different stacks

### ISSUE THREAD

**Issue by SZubarev** [aws/aws-cdk#8742]
When I have two stacks and each stack deploys layers with the same logical names CDK deploys only one lambda layer until logical names are made different. I would expect to have the layers prefixed with the stacks names (similar to Lambda) and two independent layers in this case.

### Reproduction Steps

N/A

### Error Log

No errors, just getting one layer instead of separate copies for each stack.

### Environment

  - **CLI Version      :** 1.47.0
  - **Framework Version:**
  - **Node.js Version:** v14.4.0
  - **OS               :** MacOS 10.14.6
  - **Language (Version):** TypeScript
### Other

--- 

This is :bug: Bug Report

**Comment by nija-at** [comment]
This difference in behaviour is not coming from the CDK. 

It's coming from either CloudFormation or the Lambda backend. In both cases, layers or functions, we treat them the exact same way by not passing the name property when not specified.

Perhaps this is something we can hide under the CDK and make it a little better for the user.

**Comment by nija-at** [comment]
Internal ref: tt/0503389452

**Comment by SamStephens** [comment]
Commenting; this is still a nasty little issue. Potentially it should be solved at the Cloudformation level, as this is also a problem for people using Cloudformation directly.

**Comment by madeline-k** [comment]
I agree with some prior comments in this thread, it should be solved at the Cloudformation level. It's not ideal, but you can avoid this problem by specifying the `layerVersionName` property, or by specifying different construct ids (which will result in different logical ids). 

I am going to convert this to a discussion so that people can find this info if they hit this behavior.
